test/Test/Rand/Cuts.hs

```diff
@@ -27,27 +26,30 @@ splitAtSpec = describe "splitAt" $ do
         snd (Treap.splitAt 0 smallTreap) `shouldBe` smallTreap
     it "splitAt n returns treap itself" $
         fst (Treap.splitAt 5 smallTreap) `shouldBe` smallTreap
-    it "splitAt 2 returns two treaps" $
-        bimap toList toList (Treap.splitAt 2 smallTreap) `shouldBe` ([1..2], [3..5])
+    it "splitAt 2 returns two treaps" $ do
+        let
+            (a, b) = Treap.splitAt 2 smallTreap
+        a `describedAs` ([1..2] `with` 3)
```

Comment:
Yeah, assigning proper fixity priorities to the `with` function will make code nicer 🙂